source/base/function_parser.cc

```diff
@@ -292,12 +274,9 @@ FunctionParser<dim>::init_muparser() const
     {
       fp.get().emplace_back(new mu::Parser());
 
-      for (std::map<std::string, double>::const_iterator constant =
-             constants.begin();
-           constant != constants.end();
-           ++constant)
+      for (const std::pair<std::string, double> &constant : constants)
```

Comment:
`clang-tidy` complains that the correct type for the first template parameter is (`const std::pair<const std::__cxx11::basic_string<char>, double> &`) and suggests `const auto&` instead.

Reply:
I think those are the same but lets use `auto &` to make it happy.